econml/metalearners/_metalearners.py

```diff
@@ -277,6 +279,7 @@ def fit(self, Y, T, X, *, inference=None):
         if Y.ndim == 2 and Y.shape[1] == 1:
             Y = Y.flatten()
         T = self._one_hot_encoder.fit_transform(T.reshape(-1, 1))
+        self._set_encoded_treatment_names(self._one_hot_encoder)
```

Comment:
I think the fact that this single line appears everywhere is a bit problematic and the reason why I think this logic should be going into the TreatmentExpansionMixin. For instance I believe you've forgotten to add this line to DRIV and more generally, this could become even hairier to maintain with more and more estimators.

Reply:
I refactored this and removed much of the old logic due to updates in since sklearn 0.20. Take a look!